core/src/main/java/org/elasticsearch/index/query/MissingQueryBuilder.java

```diff
@@ -78,4 +100,133 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     public String getName() {
         return NAME;
     }
+
+    @Override
+    protected Query doToQuery(QueryParseContext parseContext) throws IOException {
+        return newFilter(parseContext, fieldPattern, existence, nullValue);
+    }
+
+    public static Query newFilter(QueryParseContext parseContext, String fieldPattern, boolean existence, boolean nullValue) {
+        if (!existence && !nullValue) {
+            throw new QueryParsingException(parseContext, "missing must have either existence, or null_value, or both set to true");
+        }
```

Comment:
this needs to stay because the method can be called from any other class, it's a public static method....thus validate might not be called at all before calling this method.